From 290276687af771720de567ad309458e93ac54c95 Mon Sep 17 00:00:00 2001
From: Peter Rutschmann <peter.rutschmann@bbw.ch>
Date: Wed, 12 Apr 2023 16:00:48 +0200
Subject: [PATCH 20/27] If not exact 2 values are blank, a message has to be
 shown.

---
 src/formelrad/Formelrad.js | 60 +++++++++++++++++++++++---------------
 1 file changed, 36 insertions(+), 24 deletions(-)

diff --git a/src/formelrad/Formelrad.js b/src/formelrad/Formelrad.js
index cdd2db1..60015ec 100644
--- a/src/formelrad/Formelrad.js
+++ b/src/formelrad/Formelrad.js
@@ -15,30 +15,42 @@ export default function Formelrad() {
     const handleSubmit = (event) => {
         event.preventDefault();
         console.log("handleSubmit")
-        if (values.u === "" && values.i === "") {
-            /*calculate u and i */
-            setValues(values => ({...values, u: Math.sqrt(values.p * values.r)}));
-            setValues(values => ({...values, i: Math.sqrt(values.p / values.r)}));
-        } else if (values.u === "" && values.r === "") {
-            /*calculate u and r */
-            setValues(values => ({...values, u: values.p / values.i}));
-            setValues(values => ({...values, r: values.p / values.i / values.i}));
-        } else if (values.u === "" && values.p === "") {
-            /*calculate u and p */
-            setValues(values => ({...values, u: values.i * values.r}));
-            setValues(values => ({...values, p: values.i * values.i * values.r}));
-        } else if (values.i === "" && values.r === "") {
-            /*calculate i and r */
-            setValues(values => ({...values, i: values.p / values.u}));
-            setValues(values => ({...values, r: values.u * values.u / values.p}));
-        } else if (values.i === "" && values.p === "") {
-            /*calculate i and p */
-            setValues(values => ({...values, i: values.u / values.r}));
-            setValues(values => ({...values, p: values.u * values.u / values.r}));
-        } else {
-            /*calculate r and p */
-            setValues(values => ({...values, r: values.u / values.i}));
-            setValues(values => ({...values, p: values.u * values.i}));
+
+        let count = 0;
+        if (values.u === "") count++;
+        if (values.i === "") count++;
+        if (values.r === "") count++;
+        if (values.p === "") count++;
+        if (count !== 2) {
+            setValues(values => ({...values, message: "2 Felder leer lassen, 2 Felder ausfüllen"}));
+        }else {
+            setValues(values => ({...values, message: ""}));
+
+            if (values.u === "" && values.i === "") {
+                /*calculate u and i */
+                setValues(values => ({...values, u: Math.sqrt(values.p * values.r)}));
+                setValues(values => ({...values, i: Math.sqrt(values.p / values.r)}));
+            } else if (values.u === "" && values.r === "") {
+                /*calculate u and r */
+                setValues(values => ({...values, u: values.p / values.i}));
+                setValues(values => ({...values, r: values.p / values.i / values.i}));
+            } else if (values.u === "" && values.p === "") {
+                /*calculate u and p */
+                setValues(values => ({...values, u: values.i * values.r}));
+                setValues(values => ({...values, p: values.i * values.i * values.r}));
+            } else if (values.i === "" && values.r === "") {
+                /*calculate i and r */
+                setValues(values => ({...values, i: values.p / values.u}));
+                setValues(values => ({...values, r: values.u * values.u / values.p}));
+            } else if (values.i === "" && values.p === "") {
+                /*calculate i and p */
+                setValues(values => ({...values, i: values.u / values.r}));
+                setValues(values => ({...values, p: values.u * values.u / values.r}));
+            } else {
+                /*calculate r and p */
+                setValues(values => ({...values, r: values.u / values.i}));
+                setValues(values => ({...values, p: values.u * values.i}));
+            }
         }
     }
 
-- 
2.30.0.windows.1

