From a91e16bae0526c45a8d722cd13196bb21b0fa8fb Mon Sep 17 00:00:00 2001
From: Peter Rutschmann <peter.rutschmann@bbw.ch>
Date: Wed, 12 Apr 2023 15:21:11 +0200
Subject: [PATCH 17/27] Calculate i and p

---
 src/formelrad/Formelrad.js | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/formelrad/Formelrad.js b/src/formelrad/Formelrad.js
index fd128c3..9c3d2d9 100644
--- a/src/formelrad/Formelrad.js
+++ b/src/formelrad/Formelrad.js
@@ -10,6 +10,22 @@ export default function Formelrad() {
         r: "",
         p: ""
     })
+
+    const calculate = (event) => {
+        event.preventDefault();
+        console.log("calculate");
+        if (values.i === "" && values.r === "") {
+            /*calculate i and r */
+            setValues(values => ({...values, i: values.p / values.u}));
+            setValues(values => ({...values, r: values.u * values.u / values.p}));
+        } else if (values.i === "" && values.p === "") {
+            /*calculate i and p */
+            setValues(values => ({...values, i: values.u / values.r}));
+            setValues(values => ({...values, p: values.u * values.u / values.r}));
+            setColors( colors=> ({...colors, i:"red", p:"red"}));
+        }
+    }
+
     return (
         <>
             <section>
@@ -17,7 +33,7 @@ export default function Formelrad() {
                     <h2>Formelrad</h2>
                     <img src={formelrad} width="200" alt="Formelrad"/>
                 </header>
-                <form>
+                <form onSubmit={calculate}>
                     <InputField color={"black"} value={values.u} label="Spannung" handleChange={e => {setValues(values => ({...values, u: e.target.value}))}} />
                     <InputField color={"black"} value={values.i} label="Stromstärke" handleChange={e => {setValues(values => ({...values, i: e.target.value}))}} />
                     <InputField color={"black"} value={values.r} label="Widerstand" handleChange={e => {setValues(values => ({...values, r: e.target.value}))}} />
-- 
2.30.0.windows.1

