From 5a7fbc7f20198c6e2877495df5f2477bbebb3364 Mon Sep 17 00:00:00 2001
From: Peter Rutschmann <peter.rutschmann@bbw.ch>
Date: Wed, 12 Apr 2023 15:22:00 +0200
Subject: [PATCH 18/27] Calculate r and p

---
 src/formelrad/Formelrad.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/formelrad/Formelrad.js b/src/formelrad/Formelrad.js
index 9c3d2d9..a2afaab 100644
--- a/src/formelrad/Formelrad.js
+++ b/src/formelrad/Formelrad.js
@@ -22,7 +22,10 @@ export default function Formelrad() {
             /*calculate i and p */
             setValues(values => ({...values, i: values.u / values.r}));
             setValues(values => ({...values, p: values.u * values.u / values.r}));
-            setColors( colors=> ({...colors, i:"red", p:"red"}));
+        } else {
+            /*calculate r and p */
+            setValues(values => ({...values, r: values.u / values.i}));
+            setValues(values => ({...values, p: values.u * values.i}));
         }
     }
 
-- 
2.30.0.windows.1

